uktrade/data-hub-api
# feature/CLS2-680-include-global-account-manager-in-investment-project-index

### Description of change

Add the `one_list_group_global_account_manager` field to the investment project search index. This field comes from the investor company or its global headquarters company the project is assigned to.

The opensearch index for investment projects is reindex when a company one list account owner is updated. 

### Test instructions
You should be able to see the additional field `one_list_group_global_account_manager` when querying the Investment project on open search 
E.g. `http://localhost:8000/v3/search/investment_project`

Sample field structure:
```
------>8------
"one_list_group_global_account_manager": {
    "name": "Barry Oling",
    "last_name": "Oling",
    "id": "a80ff5fd-8904-4940-bf96-fe8047e34be5",
    "first_name": "Barry"
},
------8<------
```

There are two cases for investment projects belong to: 

1. Investor company
   - Create/select an investment project for a company.
   - Assign it an `One list account owner` through the admin.
   - This should now be listed in the `one_list_group_global_account_manager` field when queried through open search.
2. Global headquarters company
   - Create/select a subsidiary company.
   - Create/select an investment project for the subsidiary company.
   - Assign another company as its `Global headquarters`.
   - For the Global headquarters company assign an `One list account owner` through the admin.
   - The investment project of the subsidiary company should have the One list account owner assigned to the Global headquarters as the `one_list_group_global_account_manager` value when queried through open search.

### Checklist


  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>


### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.
